network/bitcoin_transaction.ksy

```diff
@@ -0,0 +1,147 @@
+    seq:
+      - id: txid
+        size: 32
+        doc: |
+          Previous transaction hash.
+      - id: output_id
+        type: u4
+        doc: |
+          ID indexing an ouput of the transaction refered by txid.
+          This output will be used as an input in the present transaction.
+      - id: script_len
+        type: u1
+        doc: |
+          ScriptSig's length.
+      - id: script_sig
+        size: script_len
+        type: script_signature
+        doc: |
+          ScriptSig.
+        doc-ref: 'https://en.bitcoin.it/wiki/Transaction#Input'
+        doc-ref: 'https://en.bitcoin.it/wiki/Script'
+      - id: end_of_vin
+        contents: [0xff, 0xff, 0xff, 0xff]
+        doc: |
+          Magic number indicating the end of the current input.
+  vout:
+    seq:
+      - id: amount
+        type: u8
+        doc: |
+          Number of Satoshis to be transfered.
+      - id: script_len
+        type: u1
+        doc: |
+          ScriptPubKey's length.
+      - id: script_pub_key
+        size: script_len
+        doc: |
+          ScriptPubKey.
+        doc-ref: 'https://en.bitcoin.it/wiki/Transaction#Output'
+        doc-ref: 'https://en.bitcoin.it/wiki/Script'
+  script_signature:
+    seq:
+      - id: sig_stack_len
+        type: u1
+      - id: der_sig
+        type: der_signature
+        doc: |
+          DER-encoded ECDSA signature.
+        doc-ref: 'https://en.wikipedia.org/wiki/X.690#DER_encoding'
+        doc-ref: 'https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm'
+      - id: sig_type
+        type: u1
+        enum: sighash_type
+        doc: |
+          Type of signature.
+      - id: pubkey_stack_len
+        type: u1
+      - id: pubkey
+        type: public_key
+        doc: |
+          Public key (bitcoin address of the recipient).
+  der_signature:
+    seq:
+      - id: sequence
+        contents: [0x30]
+      - id: sig_len
+        type: u1
+      - id: sep_1
+        contents: [0x02]
+      - id: sig_r_len
+        type: u1
+        doc: |
+          'r' value's length.
+      - id: sig_r
+        size: sig_r_len
+        doc: |
+          'r' value of the ECDSA signature.
+        doc-ref: 'https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm'
+      - id: sep_2
+        contents: [0x02]
+      - id: sig_s_len
+        type: u1
+        doc: |
+          's' value's length.
+      - id: sig_s
+        size: sig_s_len
+        doc: |
+          's' value of the ECDSA signature.
+        doc-ref: 'https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm'
+  public_key:
+    seq:
+      - id: type
+        type: u1
+      - id: x
+        size: 32
+        doc: |
+          'x' coordinate of the public key on the elliptic curve.
+      - id: y
+        size: 32
+        doc: |
+          'y' coordinate of the public key on the elliptic curve.
+enums:
+  sighash_type:
+    01: sighash_all
+    02: sighash_none
+    03: sighash_single
```

Comment:
These (01-02-03) are octal literals. This obviously does not matter in this case, but if you were not aware of that, probably trying to use `08` or `09` will trigger an error.

Reply:
The [documentation](http://doc.kaitai.io/user_guide.html#enums) talks about decimal literals for the enumerations. Is it normal?

Reply:
This has actually nothing to do with KS — it's YAML syntax. You can specify integers using 3 syntaxes:

* `123` - decimal
* `0x123`` - hexadecimal, that will be equivalent of 291 in decimal
* `0123` - octal, that will be 83 in decimal

So, if you're adding leading zeroes here for sake of beautiful formatting, please don't — it actually activates octal literals parsing, and there's nothing we can do with it in KS.

Again, I agree that this probably requires at least a word of caution in enums documentation, where you've pointed that out.

Reply:
Interesting. I didn't know that. I will change it straight away.

Reply:
Done.